compaction/compaction_backlog_manager.hh

```diff
@@ -71,6 +77,14 @@ public:
     compaction_backlog_tracker(const compaction_backlog_tracker&) = delete;
     ~compaction_backlog_tracker();
 
+    // Clone the tracker and return an unregistered, mutable copy of it.
+    future<compaction_backlog_tracker> clone_async() const;
+    // Clear the tracker gently
+    future<> clear_gently() noexcept;
+    // Swap the tracker content with another one (typically a cloned one that was changed)
+    // Tracker registration of both objects is unaffected.
+    void swap(compaction_backlog_tracker&) noexcept;
```

Comment:
Why do we need async/gently variants? How large can these grow? How long does it take to copy/clear them?

Reply:
Unfortunately, in pathological cases a compaction group may have 1000's if not 10,000's sstables.
In can measure how long it takes to clone/cleanup an oversized backlog tracker.

Reply:
thousands of tables is still peanuts (0.5ms / 1000 -> 0.5us per table, or ~2000 cycles at 2GHz).

tens of thousands is too much.

Reply:
I left that out in v3.
Will reconsider that as followup (along with sstable_set / backlog_tracker transactions)

Reply:
FWIW, we stumbled upon stalls in partitioned_sstable_set insertion and destroy.
See https://github.com/scylladb/scylla-enterprise/issues/2975

Reply:
I guess partitioned_sstable_set is very slow. It's also O(n log n), so 2000 sstables = 22000 ops.